### PR TITLE
Send `orderID` instead of `sessionID` to FPTI analytics

### PR DESCRIPTION
**Pull request by scannillo**
### Reason for changes

- Mirror [this iOS PR](https://github.com/paypal/iOS-SDK/pull/144)

### Summary of changes

- Feature clients
     - Add `AnalyticsService` instance to each feature client
         - (Since this needs an orderID, I made this a nullable param. Open to suggestions)
     - Remove clientID fetch from feature clients
 - `API`
    - Remove `sendAnalyticsEvent()` func
        - (Feature clients now interact directly with `AnalyticsService`)
    - Move fetchClientID functions into new `ClientIDAPI` layer
        - (This moves us toward eventually removing `API` in favor of `<Service>API`s per endpoint that @sshropshire proposed)
    - Remove `Context` dependency from `API`
- `APIUnitTest`
    - Move relevant tests --> `ClientIDAPIUnitTests`
    - Move relevant tests --> `AnalyticsServiceUnitTests`
- Tag changes
    - `session_id` --> `order_id`
    - `merchant_app_environment` --> `merchant_sdk_env`

Please provide feedback!

 ### Checklist

 - [x] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

**Comment by scannillo** [comment]
❓👋  Tests in `PayPalNativeClientTest.kt` are failing due to NPEs in `DeviceInspector` accessing [these](https://github.com/paypal/Android-SDK/blob/7f4541cee30fc1f1e25fa8cb784a55d7f744607b/CorePayments/src/main/java/com/paypal/android/corepayments/analytics/DeviceInspector.kt#L24) (for example: `Build.MANUFACTURER`) static properties. 

I _think_ this is because [`PayPalNativeClientTest` ](https://github.com/paypal/Android-SDK/blob/7f4541cee30fc1f1e25fa8cb784a55d7f744607b/PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt#L46)is using `mockk<Application>`, while the other test files (for example: [`CardClientUnitTest`](https://github.com/paypal/Android-SDK/blob/7f4541cee30fc1f1e25fa8cb784a55d7f744607b/CardPayments/src/test/java/com/paypal/android/cardpayments/CardClientUnitTest.kt#L55)) use `mockk<FragmentActivity>`. 

I'm confused as to why `mockk` on `Application` would cause this issue? I shouldn't have to mock `AnalyticsService` more explicitly. Would love any help on this!

**Comment by jcnoriega** [comment]
@scannillo You need to add `@RunWith(RobolectricTestRunner::class)` at the top of `PayPalNativeCheckoutTest`.

Context: `Robolectric` is an Android unit testing framework which runs your tests in a simulated Android environment. It is used when you want to test Android specific code. In this case, the `Build` class. Without it, all the `Build` variables defaulted to null, and because Kotlin is null safe, it crashed inside `DeviceInspector`, as all its variables are `not-null`. When you run with `Robolectric`, the framework populates those variables with default values.

File: PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt [hunk 1]
```diff
@@ -358,7 +354,7 @@ class PayPalNativeCheckoutClientTest {
                     assertEquals(address.countryCode, mockCountryCode)
                 }
             )
-            api.sendAnalyticsEvent("paypal-native-payments:shipping-address-changed")
+            // api.sendAnalyticsEvent("paypal-native-payments:shipping-address-changed")
```
Comment (jcnoriega) [inline PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt:360]: remove comment

File: PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt [hunk 2]
```diff
@@ -401,7 +397,7 @@ class PayPalNativeCheckoutClientTest {
                         assertTrue(option.selected)
                     }
                 )
-                api.sendAnalyticsEvent("paypal-native-payments:shipping-method-changed")
+                // api.sendAnalyticsEvent("paypal-native-payments:shipping-method-changed")
```
Comment (jcnoriega) [inline PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt:403]: remove comment

File: PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt [hunk 3]
```diff
@@ -141,7 +137,7 @@ class PayPalNativeCheckoutClientTest {
         val errorSlot = slot<PayPalSDKError>()
         val payPalCheckoutListener = spyk<PayPalNativeCheckoutListener>()
 
-        coEvery { api.fetchCachedOrRemoteClientID() } throws error
+        coEvery { mockClientIDAPI.fetchCachedOrRemoteClientID() } throws error
```
Comment (jcnoriega) [inline PayPalNativePayments/src/test/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClientTest.kt:143]: in this test, check for error code equal to 7, not 123

**Comment by scannillo** [comment]
> @scannillo You need to add @RunWith(RobolectricTestRunner::class) at the top of PayPalNativeCheckoutTest.

Thank you Jose, that is very helpful context!

**Comment by scannillo** [comment]
Chatted with Steven 🎧 about this PR. I didn't realize that https://github.com/paypal/Android-SDK/pull/133 was open with underlying `API` layer changes.

This PR does a similar thing to Steven's draft PR that is already up, by moving the clientID fetching into it's own API class.

We decided that I will re-submit this PR to be a bare-bones change, just for the sake of getting the `order_id` sent. And to leave the architectural changes to Steven's draft PR, along with a relevant ADR.

**Comment by scannillo** [comment]
Closing in favor of https://github.com/paypal/Android-SDK/pull/139

**Pull request closed without merging**